sklearn/manifold/tests/test_spectral_embedding.py

```diff
@@ -85,7 +92,8 @@ def test_sparse_graph_connected_component():
         assert_array_equal(component_1, component_2)
 
 
-def test_spectral_embedding_two_components(seed=36):
+@pytest.mark.parametrize("eigen_solver", ("arpack", "lobpcg", "amg"))
```

Comment:
Now, we can only focus on the real purpose of the test and discard the `PyAMG` issue.

```suggestion
@pytest.mark.parametrize(
    "eigen_solver",
    [
        "arpack",
        "lobpcg",
        pytest.param("amg", marks=skip_if_no_pyamg),
    ]
)
@pytest.mark.parametrize("dtype", [np.float32, np.float64])
```